core/app/models/spree/product.rb

```diff
@@ -274,6 +274,10 @@ def find_variant_property_rule(option_value_ids)
       end
     end
 
+    def cache_key
+      super + "/prices/" + prices.currently_valid.pluck(:id).join("/")
```

Comment:
Can absolutely be this one. I'm thinking after #994 is in, which gives us a first go at how a pricer would be structured, it could be quite neatly tucked into a new pricer that's less stupid than the `ConservativeLineItemPricer`.